packages/firestore/test/unit/specs/limbo_spec.test.ts

```diff
@@ -287,4 +288,49 @@ describeSpec('Limbo Documents:', [], () => {
         .expectEvents(query, { removed: [doc2] });
     }
   );
+
+  specTest(
+    'Limbo documents are resolved after primary tab failover',
+    ['multi-client'],
+    () => {
+      const query = Query.atPath(path('collection'));
+      const doc1 = doc('collection/a', 1000, { key: 'a' });
+      const doc2 = doc('collection/b', 1001, { key: 'b' });
+      const deletedDoc2 = deletedDoc('collection/b', 1005);
+
+      return (
+        client(0, false)
+          .becomeVisible()
```

Comment:
Hrm.  I thought this was here to make sure the client becomes primary, but in your other test you omit it and have .expectPrimaryState(true) instead?

Reply:
It's not really needed anymore since we drain the queue to begin with. We just needed one step (which for the first client can be a no-op).

We still need `becomeVisible` in the tests that verify that if there are any foreground tabs only foreground tabs can acquire the primary lease.